src/components/links.tsx

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { FC } from 'react';
 import Link from 'next/link';
 import { useAppSelector } from '@common/state/hooks';
 import { selectActiveNetwork } from '@common/state/network-slice';
+import { Network } from '@common/types/network';
 
-export const buildUrl = (path: string, networkMode?: string) =>
-  `${path}?chain=${encodeURIComponent(networkMode || useAppSelector(selectActiveNetwork).mode)}`;
+export const buildUrl = (path: string, network: Network) => {
+  const url = `${path}?chain=${encodeURIComponent(network?.mode)}`;
+  if (network?.isCustomNetwork) {
+    return `${url}&api=${network.url}`;
+  }
+  return url;
+};
 
-export const TxLink: React.FC<{ txid: string }> = React.memo(({ txid, ...rest }) => (
-  <Link href={buildUrl(`/txid/${encodeURIComponent(txid)}`)} passHref {...rest} />
-));
+export const StacksLink: FC<{ path: string }> = ({ path, ...rest }) => {
```

Comment:
This wrapper is nice 👍 

This is nitpicky -- but the name, at first, made me think it would be a wrapper for a stacks transaction. Do you think `ExplorerLink` might be a better name? (Hopefully code automation tools could make changing this easy 😅)